src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(
+  items: Array<string | number>,
+  key: string | number,
+): GroupsMap {
```

Comment:
The return type `GroupsMap` should be generic to match the type of the items in the array. It should be `GroupsMap<T>` to ensure type safety.